src/escapedHashtags.js

```diff
@@ -186,16 +115,47 @@ define([
     /*
      * extends xpath parser to be aware of escaped hashtags
      */
-    function parser(hashtagConfig) {
-        var xpathParser = xpath.createParser(xpath.makeXPathModels(hashtagConfig));
+    function parser(hashtagInfo) {
+        function decorateHashtagger(tagger) {
+            var super_toHashtag = tagger.toHashtag;
+            tagger.toHashtag = function (xpath_) {
+                var expr = super_toHashtag(xpath_);
+                if (expr !== null) {
+                    expr = DELIMITER + expr + DELIMITER;
+                }
+                return expr;
+            };
+            return tagger;
+        }
+        var xpathParser = xpath.createParser(xpath.makeXPathModels(hashtagInfo)),
+            escapingModels = xpath.makeXPathModels(hashtagInfo, decorateHashtagger),
+            escapingParser = xpath.createParser(escapingModels),
+            baseHashtagExpr = escapingModels.HashtagExpr;
+
+        escapingModels.HashtagExpr = function (definition) {
+            baseHashtagExpr.call(this, definition);
+            decorateHashtagger(this);
+            return this;
+        };
+
         return {
             parse: function (input) {
                 if (input.startsWith("#invalid/xpath ")) {
                     throw new Error("Invalid XPath");
                 }
-                var parsed = xpathParser.parse(toHashtag(input, xpathParser));
+                // TODO eliminate transform(input) here; should not be needed.
+                // transform should only be applied to data that is known to be
+                // escaped instead of every expression before parsing.
+                var parsed = xpathParser.parse(transform(input));
                 parsed.toEscapedHashtag = function() {
-                    return toEscapedHashtag(this.toHashtag(), xpathParser);
+                    var expr = this.toHashtag();
```

Comment:
Unfortunate that we still need to render here, then re-parse and render again below. Seems like there should be a way to pass something to `this.toHashtag()` to have it escape hashtags as it renders them (would require changes to the parser hashtag rendering code).